pkg/controller/migration/convert/network.go

```diff
@@ -437,16 +440,213 @@ func getCNIPlugin(c *components) (operatorv1.CNIPluginType, error) {
 	}
 }
 
-func handleIPPool(c *components, install *Installation) error {
+func handleIPPools(c *components, install *Installation) error {
+	pools, err := c.crdClient.IPPools().List(metav1.ListOptions{})
+	if err != nil {
```

Comment:
do we need special handling in case no pools exist?

Reply:
I don't think so.
The only addition along those lines is maybe we need some checks that we have pools for certain setups. @caseydavenport WDYT?
I think calico-ipam is covered below with the checks around assign_ip but for host-local would pools be required that we should check for?

Reply:
no need to assert on pools for anything other than Calico IPAM